### PR TITLE
add implementation of wps-rest-client with ogc-api interface (#749)

### PR DESCRIPTION
**Pull request by fmigneault**
@cehbrecht 
I made a basic equivalent of the example wps-client for OGC-API.

### CODE REVIEW

File: owslib/ogcapi/processes.py [hunk 1]
```diff
@@ -65,29 +68,31 @@ def job_list(self, process_id: str) -> dict:
...
         """
 
         path = f'processes/{process_id}/jobs'
-        return self._request_post(path, json)
+        resp = self._request_post(path, json)
+        data = resp.json()
+        return resp.headers.get("Location", data["location"])
 
-    def _request_post(self, path: str, json: dict) -> dict:
+    def _request_post(self, path: str, json: dict) -> requests.Response:
```
Comment (cehbrecht) [inline owslib/ogcapi/processes.py:86]: I would expect a json response

File: owslib/ogcapi/processes.py [hunk 2]
```diff
@@ -124,3 +129,30 @@ def result(self, process_id: str, job_id: str) -> dict:
 
         path = f'processes/{process_id}/jobs/{job_id}/results'
         return self._request(path)
+
+    def monitor_execution(self, process_id: str = None, job_id: str = None, location: str = None,
```
Comment (cehbrecht) [inline owslib/ogcapi/processes.py:133]: nice :smile:

**Review by cehbrecht — APPROVED** [review]

**Comment by cehbrecht** [comment]
@fmigneault let me know when to merge ... I will add you to the developer lists.

**Comment by cehbrecht** [comment]
Have a look at the `tests/test_ogcapi_processes_*` tests. They need to be adapted.

**Comment by cehbrecht** [comment]
@fmigneault You are free to merge now when ready. Let me know when I should do the next iteration. 

I'm looking at a different owslib topic today:

File: owslib/ogcapi/processes.py (hunk 1 above)
Comment (fmigneault) [reply owslib/ogcapi/processes.py:86]: In this case, we actually need the response object as the OGC-API specifies that the job Execution should return the Status endpoint via `Location` header. I would also prefer the JSON object, but technically we need more than the body here.

File: examples/wps-rest-client.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    DescribeProcess, Execute, GetStatus
+        -i, --identifier=[ID] process identifier - required
+    Execute
+        -d, --data, --json JSON file containing pre-made request to be submitted - required
+    GetStatus
+        -j, --job JobID returned by the Execute request - required
+
+Examples
+--------
+python wps-rest-client.py -u https://ogc-ades.crim.ca/ADES -r GetCapabilities
```
[the remark this answers is not in the document]
Comment (fmigneault) [reply examples/wps-rest-client.py:45]: We can add more examples here. That would be great in fact. `wps-client.py` also present many different example endpoints.
That endpoint is [crim-ca/weaver](https://github.com/crim-ca/weaver) but supports basic OGC-API Processes operations.

**Comment by fmigneault** [comment]
@cehbrecht 

> 
> let me know when to merge ... I will add you to the developer lists.

I updated the test to work with a few processes that are always available in Weaver. 

Good to merge whenever desired.

File: owslib/ogcapi/processes.py (hunk 1 above)
Comment (cehbrecht) [reply owslib/ogcapi/processes.py:86]: We can make our own dict which contains all relevant information. But it would be handled in `owslib`.

**Comment by cehbrecht** [comment]
merged ... I take a look at it tomorrow. I will also integrate #753.

**Comment by fmigneault** [comment]
> merged ... I take a look at it tomorrow. I will also integrate #753.

#753 shouldn't affect since `api` and `conformance` methods are the only ones used by `processes` and they are left untouched.

**Pull request merged**